kyo-core/shared/src/main/scala/kyo/Fiber.scala

```diff
@@ -36,7 +36,7 @@ import scala.util.control.NoStackTrace
   * @see
   *   [[Fiber.Unsafe]] for low-level operations requiring [[AllowUnsafe]]
   */
-opaque type Fiber[+E, +A] = IOPromise[? <: E, ? <: A]
+opaque type Fiber[+E, +A] = IOPromiseBase[E, A]
```

Comment:
The compile crashes with `[? <: E, ? <: A]`. As a workaround, I've added `IOPromiseBase` and the new `asPromise` method to cast